CONTRIBUTING.md

```diff
@@ -1,41 +1,125 @@
+and summarize the desired functionality.  Set the form field "Issue type" to "New feature".  If you have not used the
+issue tracker before you will need to register an account (free), log in, and then click on the blue "Create Issue"
+button in the top navigation bar.
+
+
+<a name="contribute-code"></a>
+
+## Contribute code/document by creating Pull Request
+Before you set out to contribute code we recommend that you familiarize yourself with the Gearpump codebase and corresponding development document at Gearpump website.
+
+_If you are interested in contributing code to Gearpump but do not know where to begin:_
+In this case you should
+[browse our issue tracker for open issues and tasks](https://issues.apache.org/jira/browse/GEARPUMP/?selectedTab=com.atlassian.jira.jira-projects-plugin:issues-panel).
+
+Contributions to the Gearpump codebase should be sent as GitHub pull requests.  See section [Create a pull request](#create-pr) below
+for details.  If there is any problem with the pull request we can iterate on it using the commenting features of
+GitHub.
+
+* For _small patches_, feel free to submit pull requests directly for those patches.
+* For _larger code contributions_, please use the following process. The idea behind this process is to prevent any
+  wasted work and catch design issues early on.
+
+    1. [Open an issue](https://issues.apache.org/jira/browse/GEARPUMP) on our issue tracker if a similar issue does not
+       exist already.  If a similar issue does exist, then you may consider participating in the work on the existing
+       issue.
+    2. Comment on the issue with your plan for implementing the issue.  Explain what pieces of the codebase you are
+       going to touch and how everything is going to fit together.
+    3. Gearpump committers will iterate with you on the design to make sure you are on the right track.
+    4. Implement your issue, create a pull request (see below), and iterate from there.
+
+
+### Contribution Guideline
+1.	For all commit, must create an issue id.
+2.	For all commit log message, must contain the issue id, Like this: fix #issueId, comments.
+3.	For all PR (pull request), the title must contains issue Id.
+4.	We use rebase and squash instead of merge to ensure the log message is clean. Check section "Pull Request merge process for Gearpump"
+5.	Every commit (except doc) must have 1 committer to review before commit.
+
+<a name="create-pr"></a>
+### Create a Pull Request
+Before working on code contribution, you need to prepare your [development environment](#build-and-test).
+
+To work on a code contribution, following process is suggested:
+
+1. You need to create a working branch
 
   ```
   git branch branch_issueId
-  git checkout branch_issusId
+  git checkout branch_issueId
   ```
 
   Work on the branch, make changes, then push to your forked gearpump repo http://github.com/<git-user_id>/gearpump
```

Comment:
I am not sure about this one. I inherit it from the previous documenet written by Sean. 

@clockfly can you verify the variable?